### ISSUE TITLE
Pylint doesn't check file if it's named exactly like the directory where the file is

### ISSUE THREAD

**Issue by goldiee36** [PyCQA/pylint#8050]
### Bug description

If I have a file in a directory, and the filename is exactly the same as the directory name, I cannot make pylint to check that file if I am not inside of that directory.

Steps to reproduce:

```sh
mkdir something
echo "import yaml" > something/something
pylint something/something 
```

Nothing happens, but if I rename the file, it works:
```sh
mv something/something something/anything
pylint something/anything 
```
Now pylint works as expected.

If I step inside the directory (cd something) and run pylint for the file, it also works.

If I create an ``__init__.py`` file in the directory, pylint starts working from outside of the directory.

I cannot figure it out how to use pylint for these kind of files.

### Command used

```shell
pylint something/something
```

### Pylint output

```shell
pylint outputs nothing
```

### Expected behavior

pylint does the check and outputs the result

### Pylint version

```shell
pylint 2.15.6
```

### OS / Environment

CentOS

**Comment by DanielNoord** [comment]
Does it also break when the file has a `.py` extension?

**Comment by paudo** [comment]
No, if the file has the `.py` it works. Also, it seems like it is case sensitive. While `pylint something/Something` and `pylint something/something.py` both work, `pylint something/something` does not.

**Comment by goldiee36** [comment]
Exactly as Paudo wrote.

**Comment by paudo** [comment]
Our rough folder structure:
```
❯ tree .
.
└── Hooks
    ├── SOMETHING_1
    │   └── SOMETHING_1 # Python Script
    └── SOMETHING_2
         └── SOMETHING_1 # Python Script
```
Some additional information:
So far I found out that if pylint is executed from within Hooks we get the unexpected behavior mentioned above where there is no output at all. If pylint is executed from one directory above (the folder in which Hooks is stored) the expected output is generated.
When I tried to debug it I noticed that in file `pylint/pylint/lint/expand_modules.py` and function `expand_modules` the execution of `path = sys.path.copy()` results in a list that includes one more entry that points to the Hooks folder via absolute path.
This leads to the file being detected as namespace with no init and not being a directory.
This finally leads to the execution of `get_module_files(os.path.dirname('<REST_OF_PATH>/Hooks/Purge_Namespace/Purge_Namespace'), ['CSV'], list_all=True)` which returns no files, since it is not detected as module.

**Comment by DanielNoord** [comment]
It's very likely that the namespace detection is broken by the fact that these files don't have a `.py` extension. That feels like a pattern that we don't directly support.
Why are the python scripts not saved as a python file?

**Comment by goldiee36** [comment]
> Why are the python scripts not saved as a python file?

The python files are part of a generic automation system where you can create a job/task, and the job automatically pulls and executes the file which is named exactly like how the job was named. The system does not require to implement the entry executable in python, and as the jobs have descriptive names like onboard_cnf, and not onboard_cnf.py or onboard_cnf.sh hence the files usually omit the extension.
